### PR TITLE
Allow draining when DaemonSet kind has custom API Group

### LINKED ISSUE #5977: Autoscaler not scaling down pods when`kind` is of the resource is same between controllers
- **Which component are you using?**: `cluster-autoscaler`
- **What version of the component are you using?**: `1.25.0`
- **What k8s version are you using?**: `v1.25.10`
- **What environment is this in?**: `kops` on AWS

We are using OpenKruise and [Advanced DaemonSet](https://openkruise.io/docs/user-manuals/advanceddaemonset/).
Autoscaler seems like detects it as regular `DaemonSet` and trying to find the corresponding `DaemonSet` for the workload, and fails. This prevents from scaling down to occur.

I'm not sure whether it's the root cause or not, but I suspect that Autoscaler doesn't respect the API group. The pods created by Advanced DaemonSet have the following `ownerReferenses`:

```yaml
  ownerReferences:
  - apiVersion: apps.kruise.io/v1alpha1
    blockOwnerDeletion: true
    controller: true
    kind: DaemonSet
    name: example-advanced-daemonset
    uid: 6f0f4fa8-2694-44df-9e68-8f00630f19c1
```

The `kind` is `DaemonSet`, but the `apiVersion` is `apps.kruise.io/v1alpha1`. So might be that the `apiVersion` is being ignored, and autoscaler just looks for regular `DaemonSet` named `example-advanced-daemonset`, which obviously doesn't exist.

Here are relevant logs from autoscaler:

```
aws-cluster-autoscaler I0724 08:09:58.813598       1 cluster.go:170] node i-0bb670250f677c0c0 cannot be removed: daemonset for devops/example-advanced-daemonset-x44c5 is not present, err: daemonset.apps "example-advanced-daemonset" not found
```

Can someone advice if this expected that the `apiVersion` is ignored or not being respected, am I missing something?

### PR DESCRIPTION
**Pull request by shamil**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

The issue is described in #5977

#### Which issue(s) this PR fixes:

Fixes #5977
Suppresses #6328

#### Special notes for your reviewer:

This PR is a continuation from #6328. Made the changes as requested by @x13n 

#### Does this PR introduce a user-facing change?

NONE

### CODE REVIEW

File: cluster-autoscaler/simulator/drainability/rules/replicacount/rule.go [hunk 1]
```diff
@@ -57,7 +58,10 @@ func (r *Rule) Drainable(drainCtx *drainability.DrainContext, pod *apiv1.Pod) dr
 	if controllerRef == nil {
 		return drainability.NewUndefinedStatus()
 	}
-	refKind := controllerRef.Kind
+
+	groupVersionKind := schema.FromAPIVersionAndKind(controllerRef.APIVersion, controllerRef.Kind)
```
Comment (vadasambar) [inline cluster-autoscaler/simulator/drainability/rules/replicacount/rule.go:62]: Docs for `FromAPIVersionAndKind` says:
```go
// FromAPIVersionAndKind returns a GVK representing the provided fields for types that
// do not use TypeMeta. This method exists to support test types and legacy serializations
// that have a distinct group and kind.
// TODO: further reduce usage of this method.
```
https://github.com/kubernetes/apimachinery/blob/e2f405af78de67e27b5dd5ceb1a3eca76331e4d5/pkg/runtime/schema/group_version.go#L296-L305

`DaemonSet` uses `TypeMeta`: https://pkg.go.dev/k8s.io/api/apps/v1#DaemonSet

Should we be using `TypeMeta`'s [`GroupVersionKind()`](https://github.com/kubernetes/apimachinery/blob/v0.29.0/pkg/apis/meta/v1/meta.go#L125) function here instead? `GroupVersionKind()` uses [`FromAPIVersionAndKInd()` internally](https://github.com/kubernetes/apimachinery/blob/3c8c1f22dc332a8a6b51033f6a7c0566b885d290/pkg/apis/meta/v1/meta.go#L125-L127) but there is less chance of `GroupVersionKind()` getting removed in the future (while `FromAPIVersionKind()` might)  

Something like (at the cost of more verbosity):
```go
v1.TypeMeta{Kind: controllerRef.Kind, APIVersion: controllerRef.APIVersion}.GroupVersionKind()
```

Comment (vadasambar) [reply cluster-autoscaler/simulator/drainability/rules/replicacount/rule.go:62]: cc: @x13n

Comment (x13n) [reply cluster-autoscaler/simulator/drainability/rules/replicacount/rule.go:62]: How about:

```
refGroup := schema.ParseGroupVersion(controllerRef.APIVersion)
refKind := controllerRef.Kind
```

`ParseGroupVersion` isn't annotated as something that is planned for removal.

Comment (vadasambar) [reply cluster-autoscaler/simulator/drainability/rules/replicacount/rule.go:62]: Sounds good to me :+1:

Comment (shamil) [reply cluster-autoscaler/simulator/drainability/rules/replicacount/rule.go:62]: Hi @vladimirvivien , @x13n 

`schema.ParseGroupVersion` returns error in addition to `schema.GroupVersion`, will the following code be sufficient?

```go
refGroup, err := schema.ParseGroupVersion(controllerRef.APIVersion)
if err != nil {
	return drainability.NewUndefinedStatus()
}

refKind := controllerRef.Kind
```

Or I should return `drainability.NewBlockedStatus` instead?

Comment (x13n) [reply cluster-autoscaler/simulator/drainability/rules/replicacount/rule.go:62]: Undefined is ok in this case. If it didn't parse, it means it wasn't a correctly formatted `APIVersion` string.

Comment (shamil) [reply cluster-autoscaler/simulator/drainability/rules/replicacount/rule.go]: Allright, changed

**Pull request merged**